pkg/linter/lintRedundantTables.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package linter
+
+import (
+	"fmt"
+
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+const redundantTablesName = "redundant tables"
+const securityGroup = "security group"
+const networkACL = "network ACL"
+
+// redundantTablesLint: tables - sgs/nacls - that no endpoint/subnet are attached to them
+type redundantTablesLint struct {
+	basicLinter
+}
+
+// a rule with the list of subnets it splits
+type nonConnectedTable struct {
+	layerName  string
+	vpcOfTable vpcmodel.VPC
+	table      vpcmodel.Filter
+}
+
+// /////////////////////////////////////////////////////////
+// lint interface implementation for filterRuleSplitSubnetLint
+// ////////////////////////////////////////////////////////
+func (lint *redundantTablesLint) lintName() string {
+	return redundantTablesName
+}
+
+func (lint *redundantTablesLint) lintDescription() string {
+	return "Access control tables for which there are no resources attached to"
+}
+
+// todo: followup https://github.com/np-guard/vpc-network-config-analyzer/issues/718
+func (lint *redundantTablesLint) check() error {
+	for _, config := range lint.configs {
+		if config.IsMultipleVPCsConfig {
+			continue // no use in executing lint on dummy vpcs
+		}
+		for _, layer := range vpcmodel.FilterLayers {
+			filterLayer := config.GetFilterTrafficResourceOfKind(layer)
+			layerName := securityGroup
+			if layer == vpcmodel.NaclLayer {
+				layerName = networkACL
+			}
+			_, _ = filterLayer, layerName
```

Comment:
what is the concept of this line?

Reply:
concept of "forget to delete it" :-)
to begin with it was to avoid compilation error in the process

Reply:
deleted